go/enclave/db/storage.go

```diff
@@ -478,6 +478,9 @@ func (s *storageImpl) StoreBatch(batch *core.Batch, receipts []*types.Receipt, d
 	if err := obscurorawdb.WriteCurrentBatchSequenceNumber(dbBatch, batch.Header.SequencerOrderNo); err != nil {
 		return fmt.Errorf("could not save the current seqencer number. Cause: %w", err)
 	}
+	if err := obscurorawdb.IncrementContractCreationCount(s.db, dbBatch, receipts); err != nil {
```

Comment:
Do we ever call StoreBatch on batches that turn out to be non-canonical? Guess we could end up double counting in those situations. Don't know if it would happen enough in practice to be relevant.

Reply:
Yeah I thought of that too, I'm not sure we can calculate accurately without some serious changes.
Think this is going to be the stepping stone for now. I will add a comment to look out for though